### PR TITLE
Trigger a mergability check on ghstack prs

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115944

Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

### CODE REVIEW

File: .github/workflows/check_mergability_ghstack.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Check mergability and dependencies for ghstack prs
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  mergability-and-dependencies-check:
```
Comment (huydhn) [inline .github/workflows/check_mergability_ghstack.yml:8]: I think that this fits better into `lint.yml` workflow.  Having a new workflow is ok, but lint seems like a natural way to keep this job

Comment (izaitsevfb) [reply .github/workflows/check_mergability_ghstack.yml:8]: One issue with `lint.yml` is that lint is triggered on pushes into main, which is redundant for mergeability (and in fact won't work for the workflow that requires PR number).

**Review by izaitsevfb — APPROVED** [review]
From the code perspective looks good. 

Let's add concurrency to the workflow (see `lint.yml` for example) and test it before merging.

File: .github/workflows/check_mergability_ghstack.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Check mergability and dependencies for ghstack prs
```
Comment (izaitsevfb) [inline .github/workflows/check_mergability_ghstack.yml:1]: Nit: typo mergability -> merg**e**ability

(here and below)

**Review by izaitsevfb — CHANGES_REQUESTED** [review]

**Comment by izaitsevfb** [comment]
Hm, I don't see the new workflow triggered.

File: .github/workflows/check_mergability_ghstack.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
```
Comment (izaitsevfb) [inline .github/workflows/check_mergability_ghstack.yml:1]:
```suggestion
name: Check mergeability and dependencies for ghstack prs
```

Comment (izaitsevfb) [reply .github/workflows/check_mergability_ghstack.yml]: Oh, and filename perhaps?

**Review by huydhn — APPROVED** [review]
LGTM!

**Review by izaitsevfb — APPROVED** [review]
Awesome, thank you! Sorry about the nits!

File: .github/workflows/check_mergeability_ghstack.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  check-regex:
+    runs-on: ubuntu-latest
+    outputs:
+      regex-match: ${{ steps.regex-match.outputs.match }}
+    steps:
+      - uses: actions/checkout@v2
```
Comment (huydhn) [inline .github/workflows/check_mergeability_ghstack.yml:13]: Look like there are some deprecation warning coming from these actions, let's update them if possible https://github.com/pytorch/pytorch/actions/runs/7227759242:

* https://github.com/actions/checkout/tree/v4
* A bit of concern that `actions-ecosystem/action-regex-match` is not actively maintained anymore https://github.com/actions-ecosystem/action-regex-match/issues/396

**Pull request closed without merging**